tests/Fixer/PhpUnit/PhpUnitExpectationFixerTest.php

```diff
@@ -240,7 +240,89 @@ public function testFoo()
     public function testBar()
     {
         $this->expectException("RuntimeException");
+        $this->expectExceptionMessageMatches("/Msg.*/");
+        $this->expectExceptionCode(123);
+        bar();
+    }
+}',
+                '<?php
+final class MyTest extends \PHPUnit_Framework_TestCase
+{
+    public function testFoo()
+    {
+        $this->setExpectedException("RuntimeException", "Msg", 123);
+        foo();
+    }
+
+    public function testBar()
+    {
+        $this->setExpectedExceptionRegExp("RuntimeException", "/Msg.*/", 123);
+        bar();
+    }
+}',
+                ['target' => PhpUnitTargetVersion::VERSION_8_4],
+            ],
+            [
+                '<?php
+final class MyTest extends \PHPUnit_Framework_TestCase
+{
+    public function testFoo()
+    {
+        $this->expectExceptionMessageMatches("/Msg.*/");
+        foo();
+    }
+}',
+                '<?php
+final class MyTest extends \PHPUnit_Framework_TestCase
+{
+    public function testFoo()
+    {
         $this->expectExceptionMessageRegExp("/Msg.*/");
+        foo();
+    }
+}',
+                ['target' => PhpUnitTargetVersion::VERSION_8_4],
+            ],
+            [
+                '<?php
+final class MyTest extends \PHPUnit_Framework_TestCase
+{
+    public function testFoo()
+    {
+        // turns wrong into wrong: has a single argument only, but ...
+        $this->expectExceptionMessageMatches("/Msg.*/");
+        $this->expectExceptionMessageMatches("fail-case");
+        foo();
+    }
+}',
+                '<?php
+final class MyTest extends \PHPUnit_Framework_TestCase
+{
+    public function testFoo()
+    {
+        // turns wrong into wrong: has a single argument only, but ...
+        $this->expectExceptionMessageRegExp("/Msg.*/", "fail-case");
```

Comment:
Why this test case? `expectExceptionMessageRegExp()` only expects one argument.

Reply:
Trivial, to show this is known.

Reply:
This way of fixing will likely make the test failing. Shouldn't we keep the behaviour? By only translating name `expectExceptionMessageRegExp` to `expectExceptionMessageMatches`, even with hundred arguments?